src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
```

Comment:
> But imo if references can be something else than an array, it should be typed that way...

This is how the vscode team has chosen to represent their IPC communication. Each end appears strongly typed, but also does a few sanity checks because ultimately it is just serialized json. It isn't really worth our time trying to debate their decisions.

The most important rule about code style is to be consistent with the code around you, unless you are willing to change all the code around you (whether that means in the function/file/repo). Since we are trying to minimize diffs to reduce conflicts, I think being consistent is the right approach unless there is an actual problem.

Being consistent has the added benefit of if/when we rebase, it will be easier to tell what has changed so we can update our custom functions to match. For example, if `provideReferences` changes, I will update `provideExternalReferences` to match.